src/memote/biomass.jl

```diff
@@ -0,0 +1,159 @@
+    x = fill(true, length(biomass_rxns))
+    for (idx, rid) in enumerate(biomass_rxns)
+        d = reaction_stoichiometry(model, rid)
+        for (m1, m2) in config.biomass.growth_metabolites
+            if m1 in ["atp", "h2o"] # must consume
+                !haskey(d, m2) && !(d[m2] < 0) && (x[idx] = false)
+            else # must produce
+                !haskey(d, m2) && !(d[m2] > 0) && (x[idx] = false)
+            end
+        end
+    end
+    return x
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+For each biomass reaction, identified by [`model_biomass_reactions`](@ref),
+calculate the molar weight of the reaction by summing the products of the
+associated metabolite coefficients with their molar masses. This sum should fall
+within [1 - 1e-3, 1 + 1e-6].
+"""
+function model_biomass_is_consistent(model; config = memote_config)
+    biomass_rxns = model_biomass_reactions(model; config)
+    to_symbol(x::String) =
+        length(x) > 1 ? Symbol(uppercase(first(x)) * x[2:end]) : Symbol(uppercase(first(x)))
+    get_molar_mass(mid) = begin
+        rs = metabolite_formula(model, mid)
+        sum(v * elements[to_symbol(k)].atomic_mass for (k, v) in rs).val
+    end
+
+    x = fill(0.0, length(biomass_rxns))
+    for (idx, rid) in enumerate(biomass_rxns)
+        d = reaction_stoichiometry(model, rid)
+        x[idx] = sum(v * get_molar_mass(k) for (k, v) in d)
+    end
+    return -x ./ 1000.0
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Check if the model can be solved under default conditions and yield a reasonable
+growth rate. Here reasonable is set via `config.biomass.minimum_growth_rate` and
+`config.biomass.maximum_growth_rate`. Optionally, pass optimization
+modifications to the solver through `config.biomass.optimizer_modifications`.
+"""
+function model_solves_default(model, optimizer; config = memote_config)
+    mu = solved_objective_value(
+        flux_balance_analysis(
+            model,
+            optimizer;
+            modifications = config.biomass.optimizer_modifications,
+        ),
+    )
+    config.biomass.minimum_growth_rate < mu < config.biomass.maximum_growth_rate
+end
+
+
+"""
+$(TYPEDSIGNATURES)
+
+Check if the model can synthesize all of the biomass precursors in all of the
+biomass functions, except those listed in `config.biomass.ignored_precursors` in
+the default medium. Set any optimizer modifications with
+`config.biomass.optimizer_modifications`.
+"""
+function model_can_produce_biomass_precursors(model, optimizer; config = memote_config)
+    stdmodel = convert(StandardModel, model) # convert to stdmodel so that reactions can be added/removed
+    biomass_rxns = model_biomass_reactions(stdmodel; config)
+    mids = Set(String[])
+    for rid in biomass_rxns
+        for mid in [k for (k, v) in reaction_stoichiometry(stdmodel, rid) if v < 0]
+            push!(mids, mid)
+        end
+    end
+    blocked_precursors = String[]
+    for mid in filter(x -> x ∉ config.biomass.ignored_precursors, mids)
```

Comment:
can we use some kind of `set_difference` function here? Or maybe better avoid allocating a vector at all and just do this below:
```
mid in config.biomass.ignored_precursors && continue
```